### PR TITLE
Setting batching as default with 8kb default value

### LINKED ISSUE #75: Use Batching by default when use throughput mode
**Description**
 Use Batching for the throughput test by default.

**TODO**:
- [x] Support for C++ Traditional
- [x] First CR [Code Review](https://github.com/rticommunity/rtiperftest/pull/76)
- [x] Support for C++ Modern
- [x] Support for Java
- [x] Support for CS
- [ ] Automatic install testing
- [x] Documentation
- [x] Full [Code Review](https://github.com/rticommunity/rtiperftest/pull/76)

### PR DESCRIPTION
**Pull request by alfonso-soto**
This pull request resolve #75

### CODE REVIEW

File: srcCpp/RTIDDSImpl.cxx [hunk 1]
```diff
@@ -384,19 +386,20 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
-            {
+            if ((i == (argc-1)) || *argv[++i] == '-') {
                 fprintf(stderr, "Missing <#bytes> after -batchSize\n");
                 return false;
             }
-            _BatchSize = strtol(argv[i], NULL, 10);
 
-            if (_BatchSize < 0 || _BatchSize > (unsigned int)MAX_SYNCHRONOUS_SIZE) {
+            if (strtol(argv[i], NULL, 10) < 0
+                    || strtol(argv[i], NULL, 10) > (unsigned int)MAX_SYNCHRONOUS_SIZE) {
```
Comment (alfonso-soto) [inline srcCpp/RTIDDSImpl.cxx]: _BatchSize is a unsigned int. Does not have sense to compare < 0.

File: srcCpp/RTIDDSImpl.cxx [hunk 2]
```diff
@@ -671,37 +683,35 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
                 _useUnbounded = MAX_BOUNDED_SEQ_SIZE;
             }
             _isLargeData = true;
-        } else if (_scan_max_size <= (unsigned long)(std::min)(MAX_SYNCHRONOUS_SIZE,MAX_BOUNDED_SEQ_SIZE)) {
+        } else {
```
Comment (alfonso-soto) [inline srcCpp/RTIDDSImpl.cxx]: No need for a 'if' statement here. We are all ready taking in a count all the range of values with 'else'

File: srcCpp/RTIDDSImpl.cxx [hunk 3]
```diff
@@ -671,37 +683,35 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
     if (_DataLen > (unsigned long)MAX_SYNCHRONOUS_SIZE) {
         fprintf(stderr, "Large data settings enabled.\n");
         _isLargeData = true;
     }
 
+    if (_isLargeData && _BatchSize > 0) {
+            fprintf(stderr, "Batching cannot be used with asynchronous writing.\n");
+            return false;
+    }
+
```
Comment (alfonso-soto) [inline srcCpp/RTIDDSImpl.cxx]: I just change the order of the if.
This already worked but it makes more sense with this order

**Comment by javmorcas** [comment]
In a different pull request: _scan_max_size doesn't follow the standard naming.

File: srcCpp/RTIDDSImpl.cxx [hunk 4]
```diff
@@ -671,37 +683,35 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
-            return false;
-        }
-    } else { // If not Scan, compare sizes of Batching and dataLen
-        /*
-         * We don't want to use batching if the sample is the same size as the batch
-         * nor if the sample is bigger (in this case we avoid the checking in the
-         * middleware).
-         */
-        if (_BatchSize > 0 && (unsigned long)_BatchSize <= _DataLen) {
+        // If not Scan, compare sizes of Batching and dataLen
```
Comment (javmorcas) [inline srcCpp/RTIDDSImpl.cxx]:
```
/*
     * If not Scan, compare sizes of Batching and dataLen
     * At this point we have checked that we are not in a latency test mode,
     * therefore we are sure we are in throughput test mode, where we do want
     * to enable batching by default in certain cases
     */
    } else if (_BatchSize > 0
                && (unsigned long)_BatchSize < _DataLen * 2
                && isBatchSizeProvided) {
        /*
            * We don't want to use batching if the batch size is not large enough
            * to contain at least two samples (in this case we avoid the checking
            * at the middleware level).
            */
        fprintf(stderr,
                "Batching disabled: BatchSize (%d) is smaller than two "
                "times the sample size (%lu).\n",
                _BatchSize,
                _DataLen);
        _BatchSize = 0;
    }
```

**Comment by javmorcas** [comment]
The merge of this issue is pending approval during RTI's Performance Meeting

File: srcCpp/RTIDDSImpl.cxx [hunk 5]
```diff
@@ -384,19 +386,20 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
             }
         } else if (IS_OPTION(argv[i], "-batchSize")) {
 
-            if ((i == (argc-1)) || *argv[++i] == '-') 
-            {
+            if ((i == (argc-1)) || *argv[++i] == '-') {
```
[the remark this answers is not in the document]
Comment (alfonso-soto) [reply srcCpp/RTIDDSImpl.cxx]: Done

File: srcCpp/RTIDDSImpl.cxx [hunk 6]
```diff
@@ -671,37 +683,42 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
+    /*
+     * If not Scan, compare sizes of Batching and dataLen
+     * At this point we have checked that we are not in a latency test mode,
+     * therefore we are sure we are in throughput test mode, where we do want
+     * to enable batching by default in certain cases
+     */
+    } else if (_BatchSize > 0 && (unsigned long)_BatchSize < _DataLen * 2) {
+            /*
+             * We don't want to use batching if the batch size is not large
+             * enough to contain at least two samples (in this case we avoid the
```
Comment (javmorcas) [inline srcCpp/RTIDDSImpl.cxx]: two samples

File: srcDoc/release_notes.rst [hunk 7]
```diff
@@ -68,6 +68,16 @@ Added RTI Perftest and RTI Connext DDS information at beginning of the test(#54)
 Starting with this release, RTI Perftest will print at the beginning of the test
 its version and the version of RTI Connext DDS used to compile against.
 
+Enable batching for Throughput-Test mode with a 8kB value (#76)
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: Enable batching for Throughput-Test mode with a 8kB value (#76)(#67)

**Comment by javmorcas** [comment]
@alfonso-soto, after discussing this with Jose, we decided that having the topic names defined in 3 different ways was kind of too much, therefore we came back to the solution of having them in the idl. Could you implement that option?

**Comment by alfonso-soto** [comment]
I finish the requested changes and other that I discuss with @javmorcas about the messages relate to batching for all the languages.

The "Batching disable" message is now on PrintConfiguration.

File: srcCpp03/perftest_cpp.h [hunk 8]
```diff
@@ -60,8 +60,9 @@ class perftest_cpp
     int Run(int argc, char *argv[]);
     bool ParseConfig(int argc, char *argv[]);
     void PrintConfiguration();
+    unsigned int GetSamplesPerBatch();
 
-  private:
+private:
```
Comment (javmorcas) [inline srcCpp03/perftest_cpp.h]: Change it back.

**Pull request merged**